src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py

```diff
@@ -747,21 +786,60 @@ def __call__(
         # 4. Preprocess image
         image = self.image_processor.preprocess(image)
 
-        # 5. set timesteps
+        # 5. Prepare timesteps
+        def denoising_value_valid(dnv):
+            return type(denoising_end) == float and 0 < dnv < 1
+
         self.scheduler.set_timesteps(num_inference_steps, device=device)
         timesteps, num_inference_steps = self.get_timesteps(num_inference_steps, strength, device)
+        timesteps, num_inference_steps = self.get_timesteps(
+            num_inference_steps, strength, device, denoising_start=denoising_start if denoising_value_valid else None
```

Comment:
`denoising_value_valid` is a function here, is this what we want? 
```suggestion
            num_inference_steps, strength, device, denoising_start=denoising_start if denoising_value_valid(denoising_start) else None
```

this PR is related too https://github.com/huggingface/diffusers/issues/6495